lua/lean/lsp.lua

```diff
@@ -33,6 +34,14 @@ function lsp.plain_term_goal(bufnr, handler)
   return vim.lsp.buf_request(bufnr, "$/lean/plainTermGoal", params, handler)
 end
 
+function lsp.handlers.definition_handler (err, method, result, client_id, bufnr, config)
+  vim.lsp.handlers['textDocument/definition'](err, method, result, client_id, bufnr, config)
+  -- auto-enable LSP for standard library files
+  if #vim.lsp.buf_get_clients(0) == 0 and vim.bo.ft == "lean" then
```

Comment:
I think checking the number of clients is better that what I did in the last commit, since we don't want to start multiple clients for the same buffer.